### ISSUE TITLE
Automatically install pygoobject for linux users

### ISSUE THREAD

**Issue by abhsag24** [NIT-dgp/UIP#176]
1.Download from source  maybe keep our own file [here](https://github.com/NIT-dgp/UIP-misc)
2.Automatic extract
3.Run setup.py install automatically.

or maybe we can upload our own version on pypi

CC @pandusonu2  maybe you could throw some light on this issue? if this is the right way to go?

**Comment by pandusonu2** [comment]
I think arch users will have better idea about this. As in Arch, the software is downloaded right from the source and installed. So @ghoshbishakh @nkprince007 Any idea?

**Comment by nkprince007** [comment]
I suppose we could use `subprocess` in `setup.py` and write up an install script on `Makefile`.
But, I guess we'd have to check distro wise for the install.

**Comment by nkprince007** [comment]
@abhsag24 Please comment on the different distros we'd be supporting.

I'm thiniking of
-[x] arch
-[x] debian

**Comment by abhsag24** [comment]
@nkprince007 Fedora as well. Also we have a setup script in the code that installs pygobject in a virtual environment, maybe we should run UIP in a virtual environment only?

**Comment by abhsag24** [comment]
because pygobject almost always conflicts with the already installed version, as its a dependency on most DE's

**Comment by nkprince007** [comment]
Couldn't we use the pre-existent version?

**Comment by abhsag24** [comment]
@nkprince007  we do that now :D

**Comment by nkprince007** [comment]
Okay, how about testing the pre-existence and installing as necessary?

**Comment by abhsag24** [comment]
we can do that. We can also just use venv :P , we have packages conflicting anyway it'd save everyone som much time if we create a venv in the make instead and no one has to worry about anything

**Comment by nkprince007** [comment]
But, that'd cause us to use venv everytime to run UIP.
That's not the safe way to do it, I feel.

**Comment by abhsag24** [comment]
@nkprince007 that could be a part of the entry script right? activate venv , then carry on?

**Comment by nkprince007** [comment]
Nope, entry script couldn't do that as no extrinsic link exists between our package and `usr/lib` installation of python sitepackages.
